WordPress/gutenberg
# [RN] Show a notice when user taps to add a disabled block option

Fixes: https://github.com/wordpress-mobile/gutenberg-mobile/issues/3055

* Gutenbert Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4005
* WPAndroid PR: https://github.com/wordpress-mobile/WordPress-Android/pull/15435
* WPiOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/17293

## Description
Whenever a block doesn't support multiple instances and is already added to
the post, its icon is greyed out and it can't be added. With this commit,
when the user taps on it, it shows a Notice with the information on why
that block can't be added.

Also contains the migration of inserter-button from a Class component to a
function component (to allow the use of Hooks).

## How has this been tested?
1.  Add the "More" block to the editor
2. Tap on the "+" to open the inserter and add a new block
3. Tap on the greyed out "More" button
4. The `Notice` should appear in the top mentioning the reason why that block can't be added





## Screenshots 

| Screenshot | Video |
| - | - |
| <img src="https://user-images.githubusercontent.com/628412/136918384-f794013f-f7fd-491a-b056-6185bd94c2de.jpg" width=350 /> | <img src="https://user-images.githubusercontent.com/628412/136918402-ba6aba3a-0aac-4fa3-9520-496f58373591.gif" width=350 /> |

## Types of changes
New feature (non-breaking change which adds functionality)





## Checklist:
